packages/prompts/src/index.ts

```diff
@@ -1014,3 +1014,482 @@ export const tasks = async (tasks: Task[], opts?: CommonOptions) => {
+	 */
+	placeholder?: string;
+}
+
+export const autocomplete = <Value>(opts: AutocompleteOptions<Value>) => {
+	// Track input, cursor position, and filtering - similar to multiselect
+	let input = '';
+	let cursor = 0;
+	let filtered = [...opts.options];
+	let selectedValue = opts.initialValue;
+	let isEventsRegistered = false;
+	let isNavigating = false;
+	
+	// Filter options based on search input
+	const filterOptions = (searchText: string) => {
+		const prevLength = filtered.length;
+		const prevSelected = filtered[cursor]?.value;
+		
+		if (searchText) {
+			filtered = opts.options.filter(option => {
+				const label = (option.label ?? String(option.value ?? '')).toLowerCase();
+				const hint = (option.hint ?? '').toLowerCase();
+				const value = String(option.value).toLowerCase();
+				const term = searchText.toLowerCase();
+				
+				return label.includes(term) || hint.includes(term) || value.includes(term);
+			});
+		} else {
+			filtered = [...opts.options];
+		}
+		
+		// If filtering changed the available options, update cursor 
+		if (prevLength !== filtered.length || !filtered.length) {
+			if (filtered.length === 0) {
+				cursor = 0;
+			} else if (prevSelected !== undefined) {
+				// Try to maintain the same selected item
+				const index = filtered.findIndex(o => o.value === prevSelected);
+				cursor = index !== -1 ? index : 0;
+			} else {
+				cursor = 0;
+			}
+		}
+		
+		// Ensure cursor is within bounds
+		if (cursor >= filtered.length && filtered.length > 0) {
+			cursor = filtered.length - 1;
+		}
+		
+		// Update selected value based on cursor
+		if (filtered.length > 0) {
+			selectedValue = filtered[cursor].value;
+		}
+	};
+	
+	// Create text prompt
+	const prompt = new TextPrompt({
+		placeholder: opts.placeholder,
+		initialValue: '',
+		input: opts.input,
+		output: opts.output,
+		render() {
+			// Register event handlers only once
+			if (!isEventsRegistered) {
+				// Handle keyboard navigation
+				this.on('key', (key) => {
+					// Start navigation mode with up/down arrows
+					if (key === 'up' || key === 'down') {
+						isNavigating = true;
+					}
+					
+					// Allow typing again when user presses any other key
+					if (key !== 'up' && key !== 'down' && key !== 'return') {
+						isNavigating = false;
+					}
+				});
+				
+				// Handle cursor movement
+				this.on('cursor', (key) => {
+					if (filtered.length === 0) return;
+					
+					// Enter navigation mode
+					isNavigating = true;
+					
+					if (key === 'up') {
+						cursor = cursor > 0 ? cursor - 1 : filtered.length - 1;
+					} else if (key === 'down') {
+						cursor = cursor < filtered.length - 1 ? cursor + 1 : 0;
+					}
+					
+					// Update selected value
+					if (filtered.length > 0) {
+						selectedValue = filtered[cursor].value;
+					}
+				});
+				
+				// Register input change handler to update filtering
+				this.on('value', () => {
+					// Only update input when not in navigation mode
```

Comment:
This doesn't seem to be working! Pressing `space` will still add to the filter value.